test/net/sourceforge/kolmafia/textui/command/GardenCommandTest.java

```diff
@@ -0,0 +1,132 @@
+package net.sourceforge.kolmafia.textui.command;
+
+import static internal.helpers.HttpClientWrapper.getRequests;
+import static internal.helpers.Networking.assertPostRequest;
+import static internal.helpers.Player.withCampgroundItem;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasSize;
+
+import internal.helpers.Cleanups;
+import internal.helpers.HttpClientWrapper;
+import net.sourceforge.kolmafia.KoLConstants.MafiaState;
+import net.sourceforge.kolmafia.StaticEntity;
+import net.sourceforge.kolmafia.objectpool.ItemPool;
+import net.sourceforge.kolmafia.request.CampgroundRequest;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class GardenCommandTest extends AbstractCommandTestBase {
+
+  public GardenCommandTest() {
+    this.command = "garden";
+  }
+
+  @BeforeEach
+  public void initializeState() {
+    HttpClientWrapper.setupFakeClient();
+    StaticEntity.setContinuationState(MafiaState.CONTINUE);
+  }
+
+  @BeforeAll
+  public static void beforeAll() {
+    // there is a leak somewhere that makes every test return "you have 0 peppermint sprouts"
```

Comment:
test_campground_fuel_asdon.html contains "otherimages/campground/pepperpatch_0.gif"
That is in AsdonMartinCommandTest.fuelValidSendsRequest()

test_visit_el_vibrato_campground.html contains "otherimages/campground/pepperpatch_1.gif"
That is in ElVibratoManagerTest.canRecogniveActivePortalAndEnergy

The latter uses withEmptyCampground.
The former does not.

Reply:
thanks